src/test/javascript/spec/component/programming-exercise/programming-exercise-plans-and-repositories-preview.component.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { MockProgrammingExerciseService } from '../../helpers/mocks/service/mock-programming-exercise.service';
+import { CheckoutDirectoriesDto } from 'app/entities/checkout-directories-dto';
+import { ProgrammingExercise, ProgrammingLanguage } from 'app/entities/programming-exercise.model';
+import { HelpIconComponent } from 'app/shared/components/help-icon.component';
+import { MockComponent } from 'ng-mocks';
+import { Subscription, of } from 'rxjs';
+import { AuxiliaryRepository } from 'app/entities/programming-exercise-auxiliary-repository-model';
+import { SimpleChanges } from '@angular/core';
+import { ProgrammingExerciseBuildPlanCheckoutDirectoriesComponent } from 'app/exercises/programming/shared/build-details/programming-exercise-build-plan-checkout-directories.component';
+import { ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent } from 'app/exercises/programming/shared/build-details/programming-exercise-repository-and-build-plan-details.component';
+
+describe('ProgrammingExerciseBuildDetailsComponent', () => {
+    let component: ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent;
+    let fixture: ComponentFixture<ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent>;
+    let programmingExerciseService: ProgrammingExerciseService;
+
+    const CHECKOUT_DIRECTORY_PREVIEW_SUBMISSION_BUILD_PLAN = '#checkout-directory-preview-submission-build-plan';
+    const CHECKOUT_DIRECTORY_PREVIEW_SOLUTION_BUILD_PLAN = '#checkout-directory-preview-solution-build-plan';
+
+    const JAVA_CHECKOUT_DIRECTORIES: CheckoutDirectoriesDto = {
+        submissionBuildPlanCheckoutDirectories: {
+            exerciseCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: '',
+        },
+        solutionBuildPlanCheckoutDirectories: {
+            solutionCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: '',
+        },
+    };
+
+    const OCAML_CHECKOUT_DIRECTORIES: CheckoutDirectoriesDto = {
+        submissionBuildPlanCheckoutDirectories: {
+            exerciseCheckoutDirectory: '/assignment',
+            solutionCheckoutDirectory: '/solution',
+            testCheckoutDirectory: 'tests',
+        },
+        solutionBuildPlanCheckoutDirectories: {
+            solutionCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: 'tests',
+        },
+    };
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent, MockComponent(HelpIconComponent), ProgrammingExerciseBuildPlanCheckoutDirectoriesComponent],
+            providers: [{ provide: ProgrammingExerciseService, useClass: MockProgrammingExerciseService }],
+        })
```

Comment:
The tests fail because the component fixture cannot be created. Here's an implementation that worked for me locally (please double-check it).

```ts
        TestBed.configureTestingModule({
            declarations: [ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent, MockComponent(HelpIconComponent), ProgrammingExerciseBuildPlanCheckoutDirectoriesComponent],
            imports: [ArtemisSharedComponentModule, ArtemisSharedCommonModule],
            providers: [
                { provide: ProgrammingExerciseService, useClass: MockProgrammingExerciseService },
                { provide: TranslateService, useClass: MockTranslateService },
            ],
        })
```

Reply:
Sorry for not waiting until the tests have passed - I should have double checked them after the conversion of the module to a standalone component!

Thanks for even investigating the needed import and providers 🙏